powertools-validation/src/test/java/software/amazon/lambda/powertools/validation/handlers/SQSWithCustomEnvelopeHandler.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.validation.handlers;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import software.amazon.lambda.powertools.validation.Validation;
+
+public class SQSWithCustomEnvelopeHandler implements RequestHandler<SQSEvent, String> {
+
+    @Override
+    @Validation(inboundSchema = "classpath:/schema_v7.json", envelope = "records[*].powertools_json(body).powertools_json(Message)")
```

Comment:
One last thing, notices how records in the envelope JMSPath starts with a lower-case `r` but that in the schema it starts with an upper-case `R`. Also, on your website documentation for builtin events, you're specifying the path with a capital `R`. When I was trying to build my custom envelope starting from the builtin events didn't really help. Also I am not sure but it's worth checking if JMSPaths are case-sensitive.

Reply:
You are actually right, the issue is that using the [aws-lambda-java-events](https://github.com/aws/aws-lambda-java-libs/tree/master/aws-lambda-java-events) library, [SQSEvent](https://github.com/aws/aws-lambda-java-libs/blob/master/aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/SQSEvent.java) has a "records" (r) field while the [true event](https://docs.aws.amazon.com/fr_fr/lambda/latest/dg/with-sqs.html) has a "Records" (R). 

The validation library uses the event library, which is actually wrong when talking about JMESPath which looks at the real JSON... The test should have `envelope = "Records[*].powertools_json(body).powertools_json(Message)"` with big R. I can change this to reflect the reality. We may end up with breaking changes. @msailes wdyt ?

Reply:
Hi Jerome,
I don't think that updating the test to have a big `R` would work, not sure maybe worth to check. I was more thinking that updating the documentation to have a small `r` would be the simpler option here.

 In any case, I don't think this is related to this PR, so I think this PR does need to be updated.

Reply:
Actually I was thinking about correcting the code to match the truth (R) and make the test with R pass. Small r should not work as the real event has big R...

Reply:
I agree with you @jeromevdl